airbyte-cdk/python/airbyte_cdk/sources/declarative/stream_slicers/datetime_stream_slicer.py

```diff
@@ -29,15 +29,19 @@ class DatetimeStreamSlicer(StreamSlicer):
     # FIXME: timezone should be declarative?
     def __init__(
         self,
-        start_datetime: MinMaxDatetime,
-        end_datetime: MinMaxDatetime,
+        start_datetime: Union[MinMaxDatetime, str],
+        end_datetime: Union[MinMaxDatetime, str],
         step: str,
         cursor_value: Union[InterpolatedString, str],
         datetime_format: str,
         config: Config,
     ):
         self._timezone = datetime.timezone.utc
         self._interpolation = JinjaInterpolation()
+        if isinstance(start_datetime, str):
+            start_datetime = MinMaxDatetime(start_datetime)
+        if isinstance(end_datetime, str):
+            end_datetime = MinMaxDatetime(end_datetime)
```

Comment:
Do we need this still? I think we settled on only accepting a MinMaxDatetime for simplicity

Reply:
fixed the factory so we don't need it anymore and updated the test https://github.com/airbytehq/airbyte/pull/14330/files#diff-852ccbc499b5f19084d81ffc40383cbcb70875737dbcafde16bf64e337fdbaafR116

Reply:
i updated this so that the factory magically instantiate the InterpolatedString if the param is set to a string
eg
`start_datetime: 2021-01-01`
but this won't work when we validate the config using json schema because it'll expect `start_datetime` to be an object, not a string.

I think this means we'll have to accept `Union[InterpolatedString, str]` everywhere we accept InterpolatedString. It'll make the code slightly uglier, but at least the schema will be clear